pallets/api/src/fungibles/mod.rs

```diff
@@ -0,0 +1,225 @@
+	#[derive(Encode, Decode, Debug, MaxEncodedLen)]
+	#[repr(u8)]
+	#[allow(clippy::unnecessary_cast)]
+	pub enum Read<T: Config> {
+		/// Total token supply for a given asset ID.
+		#[codec(index = 0)]
+		TotalSupply(AssetIdOf<T>),
+		/// Account balance for a given asset ID.
+		#[codec(index = 1)]
+		BalanceOf(AssetIdOf<T>, AccountIdOf<T>),
+		/// Allowance for a spender approved by an owner, for a given asset ID.
+		#[codec(index = 2)]
+		Allowance(AssetIdOf<T>, AccountIdOf<T>, AccountIdOf<T>),
+		/// Token name for a given asset ID.
+		#[codec(index = 8)]
+		TokenName(AssetIdOf<T>),
+		/// Token symbol for a given asset ID.
+		#[codec(index = 9)]
+		TokenSymbol(AssetIdOf<T>),
+		/// Token decimals for a given asset ID.
+		#[codec(index = 10)]
+		TokenDecimals(AssetIdOf<T>),
+	}
+
+	/// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_assets::Config<Self::AssetsInstance> {
+		/// The instance of pallet assets it is tightly coupled to.
+		type AssetsInstance;
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Transfers `value` amount of tokens from the caller's account to account `to`, with additional
+		/// `data` in unspecified format.
+		///
+		/// # Parameters
+		/// * `id` - The ID of the asset.
+		/// * `to` - The recipient account.
+		/// * `value` - The number of tokens to transfer.
+		#[pallet::call_index(3)]
+		#[pallet::weight(AssetsWeightInfoOf::<T>::transfer_keep_alive())]
+		pub fn transfer(
+			origin: OriginFor<T>,
+			id: AssetIdOf<T>,
+			target: AccountIdOf<T>,
+			amount: BalanceOf<T>,
+		) -> DispatchResult {
+			let target = T::Lookup::unlookup(target);
+			AssetsOf::<T>::transfer_keep_alive(origin, id.into(), target, amount)
+		}
+
+		/// Approves an account to spend a specified number of tokens on behalf of the caller.
+		///
+		/// # Parameters
+		/// * `id` - The ID of the asset.
+		/// * `spender` - The account that is allowed to spend the tokens.
+		/// * `value` - The number of tokens to approve.
+		#[pallet::call_index(5)]
+		#[pallet::weight(<T as Config>::WeightInfo::approve(1, 1))]
+		pub fn approve(
+			origin: OriginFor<T>,
+			id: AssetIdOf<T>,
+			spender: AccountIdOf<T>,
+			value: BalanceOf<T>,
+		) -> DispatchResultWithPostInfo {
+			let weight = |approve: u32, cancel: u32| -> Weight {
+				<T as Config>::WeightInfo::approve(cancel, approve)
+			};
+			let who = ensure_signed(origin.clone()).map_err(|e| e.with_weight(weight(0, 0)))?;
+			let current_allowance = AssetsOf::<T>::allowance(id.clone(), &who, &spender);
+			let spender = T::Lookup::unlookup(spender);
+			let id: AssetIdParameterOf<T> = id.into();
+			// If the new value is equal to the current allowance, do nothing.
+			if value == current_allowance {
+				return Ok(Some(weight(0, 0)).into());
```

Comment:
No test coverage.

Reply:
Thank you

Reply:
I asked this before but not noted, what do you use for checking test coverage? Or was this just sharp as a razor?

Reply:
RR, run tests with coverage. I will get codecov added to this repo as well so its checked via CI too.

Reply:
That would be really nice. In general, the CI needs some love. I am prioritising the remaining refactors because I want this over the finish line. If you could find the time it would be incredibly appreciated.